### PR TITLE
Fix fields ambiguity in `CommonDBTM::canUnrecurs()`

### PR DESCRIPTION
**Pull request by kabassanov**
…eld is present in the secondary table

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: src/CommonDBTM.php [hunk 1]
```diff
@@ -2556,7 +2556,7 @@ public function canUnrecurs()
                         } else {
                             // Relation based on single foreign key
                             $or_criteria[] = [
-                                $field => $this->getID(),
+                                $tablename.".".$field => $this->getID(),
```
Comment (cconard96) [inline src/CommonDBTM.php:2559]:
```suggestion
                                $tablename . "." . $field => $this->getID(),
```

**Comment by cconard96** [comment]
Is this a fix for a specific bug you noticed?

**Comment by kabassanov** [comment]
> Is this a fix for a specific bug you noticed?

Yes it is. There was an error while displaying group form:

[2023-05-09 12:14:58] glpisqllog.ERROR: DBmysql::query() in /usr/share/glpi/src/DBmysql.php line 370
  *** MySQL query error:
  SQL: SELECT COUNT(*) AS cpt FROM `glpi_problemtasks`, `glpi_tasktemplates` WHERE (((`groups_id_tech` = '1234'))) AND `glpi_problemtasks`.`tasktemplates_id` = `glpi_tasktemplates`.`id` AND  NOT (`glpi_tasktemplates`.`entities_id` IN ('
10', '19'))
  Error: Column 'groups_id_tech' in where clause is ambiguous

By the way there are some other pieces of code with the same "Relation based on single foreign key". Probably need to change them also.

**Comment by cconard96** [comment]
What group form?

**Comment by cconard96** [comment]
Issue occurs on the Group item form when the Child Entities checkbox is checked. Error is triggered during the `canUnrecurs` check.

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — CHANGES_REQUESTED** [review]
Seems OK, but same change should be done on lines 2553 and 2554.

**Comment by cconard96** [comment]
> Why not, but I'm not sure how it works exactly. I supposed that sending the item type would somehow define the table to use... Isn't it true?

No. This is just translated directly to a SQL query.
```php
$or_criteria[] = [
    $itemtype_field => $this->getType(),
    $items_id_field => $this->getID(),
];
```
becomes something like:
OR (`itemtype` = 'Computer' AND `items_id` = 32)

No guesses are made about which table the fields come from.

File: src/CommonDBTM.php [hunk 2]
```diff
@@ -2551,12 +2551,12 @@ public function canUnrecurs()
                             }
                             $or_criteria[] = [
                                 $itemtype_field => $this->getType(),
```
Comment (trasher) [inline src/CommonDBTM.php:2553]:
```suggestion
                                $tablename . "." . $itemtype_field => $this->getType(),
```

**Review by orthagh — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**